### PR TITLE
Fix some access levels in `MainSolver` and add some checks to public …

### PR DESCRIPTION
**Pull request by Tomaqa**
…member functions.

Related to #688 (does not fully resolve)

### CODE REVIEW

**Comment by Tomaqa** [comment]
Making `getSMTSolver` protected and also `THandler` is not possible at the moment, some classes such as `BitBlaster` are using it. I am afraid it would require more thorough refactoring.

File: test/unit/test_UFInterpolation.cc [hunk 1]
```diff
@@ -11,6 +11,12 @@ bool verifyInterpolant(PTRef itp, PartitionManager & pManager, ipartitions_t con
     return VerificationUtils(logic).verifyInterpolantInternal(pManager.getPartition(Amask, PartitionManager::part::A), pManager.getPartition(Amask, PartitionManager::part::B), itp);
 }
 
+struct TestMainSolver : MainSolver {
```
Comment (blishko) [inline test/unit/test_UFInterpolation.cc:14]: I don't particularly like the approach of creating subclasses solely for the purpose being able to access some protected member.

I think a better solution here would be to do what `LRAInterpolationTest` is doing, i.e., explicitly track the partitions in each test.
That way the public API is sufficient.

**Review by blishko — COMMENTED** [review]
The changes in MainSolver are good, but I suggest to deal with `UFInterpolationTest` differently and make it consistent with how `LRAInterpolationTest` works.

**Review by blishko — APPROVED** [review]
LGTM!

**Comment by blishko** [comment]
Please, next time, make sure that the title of git commit is limited to 72 characters, so that GitHub does not split it when creating a PR.
Having the sentence split between title and description like in this PR is weird.

**Pull request merged**